go-controller/pkg/util/net_linux_unit_test.go

```diff
@@ -0,0 +1,951 @@
+			onRetArgsNetLinkLibOpers: []onCallReturnArgs{
+				{"AddrAdd", []string{"*mocks.Link", "*netlink.Addr"}, []interface{}{nil}},
+			},
+		},
+	}
+	for i, tc := range tests {
+		t.Run(fmt.Sprintf("%d:%s", i, tc.desc), func(t *testing.T) {
+
+			for _, item := range tc.onRetArgsNetLinkLibOpers {
+				call := mockNetLinkLibOperations.On(item.onCallMethodName)
+				for _, arg := range item.onCallMethodArgType {
+					call.Arguments = append(call.Arguments, mock.AnythingOfType(arg))
+				}
+				for _, ret := range item.retArgList {
+					call.ReturnArguments = append(call.ReturnArguments, ret)
+				}
+				call.Once()
+			}
+
+			for _, item := range tc.onRetArgsLinkIfaceOpers {
+				call := mockLink.On(item.onCallMethodName)
+				for _, arg := range item.onCallMethodArgType {
+					call.Arguments = append(call.Arguments, mock.AnythingOfType(arg))
+				}
+				for _, ret := range item.retArgList {
+					call.ReturnArguments = append(call.ReturnArguments, ret)
+				}
+				call.Once()
+			}
+			err := LinkAddrAdd(tc.inputLink, tc.inputNewAddr)
+			t.Log(err)
+			if tc.errExp {
+				assert.Error(t, err)
+			}
+			mockNetLinkLibOperations.AssertExpectations(t)
+			mockLink.AssertExpectations(t)
+		})
+	}
+}
+
+func TestLinkRoutesDel(t *testing.T) {
+	mockNetLinkLibOperations := new(mocks.NetLinkLibOperations)
+	mockLink := new(netlink_mocks.Link)
+	// below is defined in net_linux.go
+	netLinkOperations = mockNetLinkLibOperations
+
+	tests := []struct {
+		desc                     string
+		inputLink                netlink.Link
+		inputSubnets             []*net.IPNet
+		errExp                   bool
+		onRetArgsNetLinkLibOpers []onCallReturnArgs
+		onRetArgsLinkIfaceOpers  []onCallReturnArgs
+	}{
+		// TODO: Code may need to be updated to check that link is not-nil to avoid panic
+		/*{
+			desc: "link is nil",
+			inputLink: nil,
+			inputSubnets: nil,
+			errExp: true,
+			onRetArgsNetLinkLibOpers: []onCallReturnArgs{
+				{"RouteList",[]string{"*mocks.Link","int"},[]interface{}{netlink.Route{}, fmt.Errorf("mock error")}},
+			},
+		},*/
+		{
+			desc:      "fails to get routes for link",
+			inputLink: mockLink,
+			errExp:    true,
+			onRetArgsNetLinkLibOpers: []onCallReturnArgs{
+				{"RouteList", []string{"*mocks.Link", "int"}, []interface{}{[]netlink.Route{}, fmt.Errorf("mock error")}},
+			},
+			onRetArgsLinkIfaceOpers: []onCallReturnArgs{
+				{"Attrs", []string{}, []interface{}{&netlink.LinkAttrs{Name: "testIfaceName"}}},
+			},
+		},
+		{
+			desc:         "subnet input is nil and error returned is nil",
+			inputLink:    mockLink,
+			inputSubnets: nil,
+			onRetArgsNetLinkLibOpers: []onCallReturnArgs{
+				{"RouteList", []string{"*mocks.Link", "int"}, []interface{}{[]netlink.Route{}, nil}},
+			},
+		},
+		{
+			desc:         "route delete fails",
+			inputLink:    mockLink,
+			inputSubnets: ovntest.MustParseIPNets("10.18.20.172/24", "192.168.1.12/24"),
+			onRetArgsNetLinkLibOpers: []onCallReturnArgs{
```

Comment:
are you missing `errExp: true`?

Reply:
I believe so, let me confirm and fix accordingly